devlib/native/win_native.cpp

```diff
@@ -168,6 +169,53 @@ namespace winutil {
     }
 
 
+    static auto findBusNumber(DEVINST const& handle, QMap<QString, int> & cachedDeviceBuses) {
+        static auto rootHubsBuses = initializeRootBuses();
+        qDebug() << "Buses" << rootHubsBuses;
+
+
+        auto currentDevInst = handle;
+        DEVINST parentDevInst;
+        WCHAR instanceID [MAX_DEVICE_ID_LEN];
+
+        auto visitedDevices = QSet<QString>{};
+        CM_Get_Device_ID(currentDevInst, instanceID , MAX_PATH, 0);
+        visitedDevices.insert(QString::fromWCharArray(instanceID));
+
+        int bus;
+        for(;;) {
+            if (CM_Get_Parent(&parentDevInst, currentDevInst, 0) != CR_SUCCESS) {
+                bus = -1;
+                break;
+            }
+            if (CM_Get_Device_ID(parentDevInst, instanceID , MAX_PATH, 0) != CR_SUCCESS) {
+                bus = -2;
```

Comment:
You could optionally name these "error constants" -1 and -2

Reply:
Done